### PR TITLE
Expodentialy improve the speed of the forcefield

### PR DESCRIPTION
**Pull request by Techcable**
Currently the anti-safezoning is implemented as so:
If you consider the forcefield radius a circle, the old implementation checks every block in the radius of the circle to see if borders a no-pvp region. The cost of this implementation increases with the square of the forcefield radius.
My implementation calculates the border blocks and startup, and checks if each border block is in the radius of the circle.
Even assuming it is the same performance cost to check for a border, and to see if a location is inside the radius of a circle ,  my implementation is more performant.
Assuming a 32x32x20 region, with a forcefield radius of 100, here are how many checks each implemetnation does:
old: 100 radius creates a circle with PI times 100^2 which is 31415 checks.
new: 32x32x20 region has a parameter of 5120, so 5120 checks.
When you consider that checking if a block is inside a circle is _far_ more efficient than checking if a block borders the region, the performance advantage increases even more.
TLDR: Old == O(radius^2) New == O(surface area of region)

The new implementation is asynchronous, which literally removes it from the timings.

---

Similar technology has been tested and proven in my plugin [SpawnShield](https://github.com/TechzoneMC/SpawnShield).

### CODE REVIEW

**Comment by Byteflux** [comment]
Thanks, I decided to take an alternative approach here: d7325ed177d3a3496c6080884998d5cb29baf61d

Some things to note about your implementation:
- Effective radius appears much smaller than configured radius
- Forcefield updates seem very delayed in comparison to movement
- Forcefield is not rectangular
